client/my-sites/plugins/plugin-meta/index.jsx

```diff
@@ -424,19 +426,39 @@ const PluginMeta = React.createClass( {
 		analytics.ga.recordEvent( 'Plugins', 'Clicked Update All Sites Plugin', 'Plugin Name', this.props.pluginSlug );
 	},
 
+	getExtensionPath( plugin ) {
+		const pluginSlug = get( plugin, 'slug' );
+		const sections = sectionsModule.get();
+		const index = findIndex( sections, { name: pluginSlug } );
+
+		if ( ( index === -1 ) || ! includes( sections[ index ].envId, config( 'env_id' ) ) ) {
+			return '';
+		}
+
+		return get( sections[ index ], 'paths', [] )[ 0 ];
+	},
+
 	render() {
 		const cardClasses = classNames( 'plugin-meta__information', {
 			'has-button': this.hasOrgInstallButton(),
 			'has-site': !! this.props.selectedSite,
 			'is-placeholder': !! this.props.isPlaceholder
 		} );
+		let target = '_blank';
 
 		const plugin = this.props.selectedSite && this.props.sites[ 0 ] ? this.props.sites[ 0 ].plugin : this.props.plugin;
 		let actionLinks = get( plugin, 'action_links' );
 		if ( get( plugin, 'active' ) && isEmpty( actionLinks ) ) {
 			actionLinks = this.getDefaultActionLinks( plugin );
 		}
 
+		const path = this.getExtensionPath( plugin );
+
+		if ( path && has( actionLinks, 'Settings' ) ) {
```

Comment:
diff that wfm:

```diff
diff --git a/client/my-sites/plugins/plugin-meta/index.jsx b/client/my-sites/plugins/plugin-meta/index.jsx
index 03d71e2..80527e6 100644
--- a/client/my-sites/plugins/plugin-meta/index.jsx
+++ b/client/my-sites/plugins/plugin-meta/index.jsx
@@ -454,8 +454,9 @@ const PluginMeta = React.createClass( {
 
                const path = this.getExtensionPath( plugin );
 
-               if ( path && has( actionLinks, 'Settings' ) ) {
-                       actionLinks.Settings = `${ path }/${ this.props.slug }`;
+               const translatedSettings = this.props.translate( 'Settings' );
+               if ( path && has( actionLinks, translatedSettings ) ) {
+                       actionLinks[ translatedSettings ] = `${ path }/${ this.props.slug }`;
                        target = '_self';
                }
 
```

But do we even want to take this route? I mean the more relevant factor is that there _is_ a matching Calypso extension, so maybe tying that to the presence of a 'Settings' link in wp-admin can be deemed a bit arbitrary (since e.g. WPJM doesn't seem to have one, as I can imagine a number of CPT-heavy plugins don't).

What about just adding a button regardless? Ideally visually distinct from the `actionLinks` buttons that point to wp-admin. Or maybe just turn the plugin title into a link? /cc @rickybanister for design input